### DISCUSSION TITLE
I accidentally merged in github a branch i didnt mean to how do i revert it?

### DISCUSSION THREAD

**Discussion by Sashanovg** [community/community#23122] [Projects and Issues]
<p>please help I’m trying to figure it out and I cant do anything</p>

**Comment by Sashanovg** [comment, marked as answer]
<p>apparently you can just revert it through github<br>
<a href="https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/reverting-a-pull-request" class="onebox" target="_blank" rel="noopener nofollow ugc">https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/reverting-a-pull-request</a></p>

**Comment by jsoref** [comment]
<p>Odds are you don’t want to use a <code>revert</code> function for accidentally merging a branch. You generally would want to use a <code>force-push-with-lease</code> which will wipe out the merge <strong>and all subsequent commits</strong>. Yes, doing it this way is a bit riskier at the time you do it, but it’s a lot cleaner for anyone looking at the project’s history later.</p>
<p>To do the right thing would involve cherry-picking any subsequent commits (if there are any, but hopefully you didn’t make any additional commits).</p>
<p>There are many problems w/ accidentally merging a branch. Assuming you haven’t made a release from the code, it’s generally a better story to tell future developers that you didn’t include that code yet, not that you included the code and then removed the code.</p>
